rasa_sdk/forms.py

```diff
@@ -567,7 +578,10 @@ async def _validate_if_required(
             if utils.is_coroutine_action(self.validate):
                 return await self.validate(dispatcher, tracker, domain)
             else:
-                return self.validate(dispatcher, tracker, domain)
+                # see https://github.com/python/mypy/issues/5206
```

Comment:
if we have to do this a lot in rasa, we might consider contributing to mypy to support useful type guards

Reply:
yes this sounds like a good idea :+1: maybe something for someone from the team to contribute?

Reply:
yes definitely. I will keep this as a note for now, and if this comes back I'll mention it (in backend eng meeting for instance)